inabox/tests/setup_test_harness.go

```diff
@@ -117,10 +127,14 @@ func NewTestHarnessWithSetup(infra *InfrastructureHarness) (*TestHarness, error)
 	}
 
 	// Setup payload disperser
-	if err := setupPayloadDisperserForContext(testCtx, infra); err != nil {
+	if err := setupPayloadDisperserForContext(ctx, testCtx, infra); err != nil {
```

Comment:
ok so imo (havent checked all the dependencies and whether this works cleanly given current state of inabox), clients should NOT be part of the inabox infra. They should be built on a per-test basis when needed.

This is the right separation of concern imo. This way you can control your payment tests, especially parallelization. Run 4 tests in parallel, they each start by creating a new reservation, and then build the clients using that private key.

Reply:
Definitely on board. 

As I've mentioned elsewhere, I feel like our infra setup should be split in to multiple parts _in general_. But it's especially the case for clients, that they shouldn't be part of the "infra".

I don't think this will be hugely difficult to pull off, but there will be some level of reorganization, which I think exceeds the amount of inabox tweaks that fit into this PR.